src/content/blog/drupal7-drupal9-migration-ddev-acquia-migrate-accelerate.md

```diff
@@ -0,0 +1,328 @@
+
+Wow! My Drupal 7 site is already working in DDEV! Open the URL and check it out - 
+[https://d7ama-www.ddev.site](https://d7ama-www.ddev.site)
+
+### Install and configure the Drupal 9 site to prepare for AM:A
+
+We will use the conveniently included version of [ACLI](https://docs.acquia.com/acquia-cli/) within the DDEV `web` container to generate a new Drupal 9 site (*you can also omit all parameters and be prompted for answers instead if you prefer*). The important consideration, for DDEV, is explicitly using `php8.1` to run this `acli` command. Otherwise, the `acli` command will fail to run on the currently configured version of `php7.4`. 
+
+Run the following command:
+
+```
+ddev exec php8.1 /usr/local/bin/acli \
+app:new:from:drupal7 \
+--drupal7-directory=/var/www/html/d7 \
+--directory=/var/www/html/d9
+```
+
+![new D9 site has been scaffolded with the help of acli](/img/blog/2023/11/ddev-ama-blog-2-d9-acli-created.png)
+
+The new Drupal 9 site scaffolding has been created with information from *your* Drupal 7 site!
+
+<mark>**Important**</mark> - SSH into the DDEV container to install the D9 site.
+
+```
+ddev ssh
+```
+
+**The command above logs you into the <mark>`/var/www/html/d7`</mark> directory, you need to change to the <mark>`/var/www/html/d9`</mark> directory** to accomplish the following: 
+
+- Install the D9 site within our single DDEV project. 
+- Create the D9 DDEV database named: `dbd9`.
+- Configure the D9 `settings.php` with the appropriate database credentials.
+
+```
+cd ../d9
+
+vendor/bin/drush site-install \
+    --account-name=admin \
+    --account-pass=admin \
+    --site-name="example.com on D9 thanks to AM:A" \
+    --no-interaction \
+    --db-url=mysql://root:root@db/dbd9
+```
+
+![Drupal 9 site installed and configured ](/img/blog/2023/11/ddev-ama-blog-3-site-install-complete.png)
+
+The new D9 site has been `site-install`ed!
+
+
+### Install and configure Acquia Migrate: Accelerate (AM:A)
+
+Initiate the AM:A magic with the following commands <mark>***(remember we are still within the `ssh` session of the DDEV project)***</mark>. *(Note for clarity: we are running the **D9** included version of `vendor/bin/drush` on the **D9** site. Running `drush` otherwise or elsewhere would affect the **D7** site.)*
+
+```
+jq -r '.installModules[]' < acli-generated-project-metadata.json | xargs php -d memory_limit=512M vendor/bin/drush pm:install -y
+```
+*(the above installs all modules whose migration recommendations have been vetted, to ensure those migrations are available out of the box)*
+```
+vendor/bin/drush state:set --input-format=json acquia_migrate.initial_info - < acli-generated-project-metadata.json
+```
+*(the above provides all the metadata to the AM:A Drupal module's Module Auditor UI by storing it in Drupal's state )*
+
+### <mark>Logout of the `ddev ssh` session</mark>, and modify the DDEV project to serve a second URL for the d9 site:
+
+```
+logout
```

Comment:
I didn't even know this existed. Always use `exit` or `<ctrl-d>` - Learn something every day!

Reply:
I remember being taught that `logout` was the best practice back in the days of the SunOS systems I was working on decades ago, and it's stuck with me since. 😁